.github/workflows/dockerfile-push.yml

```diff
@@ -24,8 +22,25 @@ jobs:
         username: ${{ secrets.DOCKERHUB_USERNAME }}
         password: ${{ secrets.DOCKERHUB_TOKEN }}
 
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
 
+    # Cache Docker layers
+    - name: Cache Docker layers
+      uses: actions/cache@v3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
+
+    # Build and push Docker image
     - name: Build and push Dockerfile
-      run: |
-        docker build . --file docker/Dockerfile --tag unifyai/ivy:latest
-        docker push unifyai/ivy:latest
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        file: docker/Dockerfile
+        push: true
+        tags: unifyai/ivy:latest
+        cache-from: type=local,src=/tmp/.buildx-cache
```

Comment:
Difference between gha cache and local cache. Can you explain the difference between them? The original doc doesn't seem to have lot of information on how to use these variables.

Reply:
GitHub Actions Cache (GHA Cache)
What is it? It's a storage space that GitHub provides. We use it to keep things we frequently use, so we don't have to fetch or recreate them every time our automation scripts (or "workflows") run.
Where is it? This cache is on GitHub's servers, which means it's remote and not on the same machine where our tasks actually run.
When do we use it? Mostly for things that don't change often but take time to fetch or build, like certain software packages or, in our case, parts of Docker images.
Any limits? Yes, it's like a storage locker – we can't store more than a certain amount, and if we don’t access it for a while, GitHub may clear it.

Local Cache
What is it? It's also a storage space but on the machine (or "runner") that's doing our tasks, not on GitHub's servers.
Where is it? Right on the same computer or "runner" where our GitHub Action is happening.
When do we use it? It's useful if we're doing several tasks in a row in the same workflow, and we don't want to redo the same steps over and over.
Any limits? The main thing is, it's temporary. Imagine it like RAM on a computer. If the machine restarts (which GitHub does often with their machines), everything in this cache is gone.

Reply:
We have to use github cache, we want this cached image to be available in all github workflows. Just to confirm here: we are caching the dockerfile right?